src/applications/disability-benefits/526EZ/pages/primaryAddress.js

```diff
@@ -558,13 +527,53 @@ export const uiSchema = {
         'ui:options': {
           expandUnder: 'view:hasForwardingAddress'
         },
+        'ui:order': [
+          'effectiveDate',
+          'country',
+          'addressLine1',
+          'addressLine2',
+          'addressLine3',
+          'city',
+          'state',
+          'zipCode'
+        ],
+        effectiveDate: _.merge(
+          {},
+          dateUI('Effective date'),
+          { 'ui:required': ({ veteran }) => (hasForwardingAddress(veteran)) }
+        ),
         country: {
```

Comment:
Can we now use the common address uiSchema?

Reply:
Maybe? I hadn't really explored that option because the schemas are different (mainly due to regex differences on all fields), and the validations / required fields are different as well. There's also the matter of the state list eventually getting populated by the API, whereas the common UI schema does some fancy work to change the list depending on the country (USA / MEX / CAN). I'd be happy to chat about this later - my gut feeling is that I'd end up spending more time trying to get the common schema to work for us than if I just make a new uiSchema for our addresses. I could definitely be wrong though.

Reply:
IIRC @jbalboni's recommendation was to make use of the common schema logic if possible, perhaps we could use _.set to update whatever needs to be extended (validations, requirements, etc.)?

Reply:
Thanks @cvalarida can you please indicate which fields would be required but hidden using the common schema? I think @jbalboni recommended using the common schema because it provides the basic UI functionality we need here (besides some of the common validations @markgreenburg mentioned, which we could possibly add via _.set or somesuch), as well as some additional assistance for the user (when they select Canada or Mexico), but I'll let him chime in if that's not what he had in mind.

Reply:
Thanks Mark, I've added a list of the discrepancies to the issue to facilitate discussion on this:  https://github.com/department-of-veterans-affairs/vets.gov-team/issues/10595#issuecomment-396294475